dataactcore/models/baseInterface.py

```diff
@@ -161,8 +144,32 @@ def getNameFromDict(self, model, dictName, fieldName, fieldValue, idField):
         raise ValueError("Value: " + str(fieldValue) + " not found in dict: " + str(dict))
 
 
+def dbConnection():
+    """Use the config to set up a database engine and connection"""
+    if not CONFIG_DB:
+        raise ValueError("Database configuration is not defined")
+
+    dbName = CONFIG_DB['db_name']
+    if not dbName:
+        raise ValueError("Need dbName defined")
+
+    # Create sqlalchemy connection and session
+    engine = sqlalchemy.create_engine(
+        "postgresql://{username}:{password}@{host}:{port}/{db_name}".format(
+            **CONFIG_DB),
+        pool_size=100, max_overflow=50)
+    connection = engine.connect()
+    return engine, connection
+
+
 @contextmanager
 def databaseSession():
-    interface = BaseInterface()
-    yield interface.session
-    interface.close()
+    engine, connection = dbConnection()
+    sessionMaker = scoped_session(sessionmaker(
+        bind=engine, autoflush=True))
+    session = sessionMaker()
+    yield session
```

Comment:
Checking myself on how this works...so anything after `yield session` automatically gets done after the code in the invoking (for lack of a better word) `with` gets executed?